### ISSUE TITLE
Full logging of actions

### ISSUE THREAD

**Issue by jbouzanquet** [CSULA-Qdots/filter#13]
It could be valuable to know what's touched our data. Currently the filter program has good logging, not the other software. This issue can keep track of what we need to do with respect to records of data operations.
- [x] baselineCorrector.sh
- [x] filter.tcl
- [x] rescale.tcl
- [ ] zero.m

There is a suggestion of a logging method at 
http://stackoverflow.com/questions/1518072/suppress-start-message-of-matlab

**Comment by jbouzanquet** [comment]
I implemented logging within baselineCorrector, so doing so within zero would be redundant. It might be useful though if the log from filter were renamed to log-filter-date instead of log-date, as the new logs go by the name log-baseline-date.

**Comment by blackdjinn** [comment]
Good idea. Change made and pushed to repo.

**Comment by blackdjinn** [comment]
Oh sweet... I didn't know it picked up on the mention in the commit message.

So.. my thoughts on logging in general:
It should contain enough information that the entire process could be reconstructed by hand if needed. This means things like software versions, command line parameter, etc should end up in the log.

**Comment by jbouzanquet** [comment]
Makes sense. I could add a version to the scripts I'm making right now. 

Did you read the page on github flavored markdown? It has several neat features like the automatic linking and tasklists.

**Comment by blackdjinn** [comment]
I hadn't gone into it, no... Tasklists... OK that needs looking at. I was commenting on the bit where it picked up that I pushed a commit up that referenced the issue, and it showed up in the conversation log here.

**Comment by jbouzanquet** [comment]
So the m files could have a version as well, reported in the log of what uses it. A simple format we could follow is making the first line of all the m files be `% Name Version#`. The program calling it could head -n1 the m file into the log to report the version of math tools used.

**Comment by blackdjinn** [comment]
That could work. I think I'm just stapling the output of 'git show' and 'git ststus' to the logs on the filter... this has the nice benefit of letting us know exactly which commit and if there are any changes pending.

**Comment by jbouzanquet** [comment]
That's a good idea that I think only works in the development directory. Everyone has to add that to their shell $path and Matlab path, and even then I'm not entirely sure the cwd for the git command in the script would be the one the script lives in. I've been using the symlink-dropbox interaction to put a copy of the scripts we use in each directory we analyse data in and keep the copies updated. When a script runs from qd/data/out/eV, won't the git status fail? We know the m files will be in the same directory the scripts run from because they don't work otherwise.

Dropbox's way of handling symlinks means you all see a file, but when my computer accesses Dropbox the files everyone else sees update to match the linked one. Here's a list of all the symlinks I've made (current directory is QD/data)

```
lrwxrwxrwx   27 Mar  1 15:35 ./baselineCorrector.sh -> filter/baselineCorrector.sh
lrwxrwxrwx   17 Mar  1 15:36 ./filter.tcl -> filter/filter.tcl
lrwxrwxrwx   13 Mar  1 17:50 ./generatedData/filter.tcl -> ../filter.tcl
lrwxrwxrwx    8 Mar  1 17:50 ./generatedData/ipf.m -> ../ipf.m
lrwxrwxrwx   12 Mar  8 13:42 ./ipf.m -> filter/ipf.m
lrwxrwxrwx   26 Mar  1 15:40 ./out/eV/baselineCorrector.sh -> ../../baselineCorrector.sh
lrwxrwxrwx   11 Feb  7 15:03 ./out/eV/ipf.m -> ../../ipf.m
lrwxrwxrwx   12 Mar  1 16:03 ./out/eV/zero.m -> ../../zero.m
lrwxrwxrwx   11 Feb  7 15:04 ./out/WL/ipf.m -> ../../ipf.m
lrwxrwxrwx   13 Mar  1 15:35 ./zero.m -> filter/zero.m

```

**Comment by blackdjinn** [comment]
OK... We are already dropping support for non-Linux systems, especially since the other folks can run an emulated session.

We can have a standard installation that makes symlinks in /usr/local places. So... clone dev tree, run install script... it works, right?

**Comment by blackdjinn** [comment]
Only thing that might make a difference on this would be having the ability to pass a logfile name option to Fiter, but... that would be non-trivial. Could have it hand the filename out in a file if wanted tho, so it could be used by other stuff for logging.. keep the whole run's log in the same output?

**Comment by jbouzanquet** [comment]
I can't think of any nice way to pass one special filename among many between shell scripts, we should probably just leave it as separate zeroing and filtering logs.
